Upstream: Reported, https://sourceforge.net/p/yoshimi/patches/8/

diff -Naur src/CMakeLists.txt src/CMakeLists.txt
--- src/CMakeLists.txt	2016-02-28 14:56:14.000000000 +0100
+++ src/CMakeLists.txt	2016-03-10 12:45:00.355743847 +0100
@@ -35,6 +35,7 @@
 message (STATUS "Building Yoshimi version ${YOSHIMI_VERSION} for ${CMAKE_SYSTEM_NAME}")
 
 include (CheckCSourceCompiles)
+include(GNUInstallDirs)
 
 mark_as_advanced (EXECUTABLE_OUTPUT_PATH)
 mark_as_advanced (LIBRARY_OUTPUT_PATH)
@@ -455,7 +456,7 @@
 
 install (TARGETS yoshimi RUNTIME DESTINATION bin)
 
-install (DIRECTORY ../banks DESTINATION share/yoshimi
+install (DIRECTORY ../banks DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -466,7 +467,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../examples DESTINATION share/yoshimi
+install (DIRECTORY ../examples DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -477,7 +478,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../presets DESTINATION share/yoshimi
+install (DIRECTORY ../presets DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -488,7 +489,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../doc/ DESTINATION share/doc/yoshimi
+install (DIRECTORY ../doc/ DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/yoshimi-${YOSHIMI_VERSION}
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -500,17 +501,17 @@
 )
 
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/yoshimi.desktop
-    DESTINATION share/applications)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.png
-    DESTINATION share/pixmaps)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pixmaps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.svg
-    DESTINATION share/icons)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi_alt.svg
-    DESTINATION share/icons)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.appdata.xml
-    DESTINATION share/appdata)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/appdata)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.1
-    DESTINATION share/man/man1)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/man/man1)
 
 set_directory_properties (PROPERTIES
     ADDITIONAL_MAKE_CLEAN_FILES "${FltkUI_headers}"
@@ -540,8 +541,6 @@
 
 if (BuildLV2Plugin)
 
-include(GNUInstallDirs)
-
 if (EXISTS "/etc/debian_version")
     set(LV2_INSTALL_DIR lib/lv2 CACHE PATH "Specifies where the LV2 libraries should be installed")
 else (EXISTS "/etc/debian_version")
