Upstream: Reported, https://sourceforge.net/p/yoshimi/patches/8/

diff -Naur src/CMakeLists.txt src/CMakeLists.txt
--- src/CMakeLists.txt	2015-07-11 20:08:16.000000000 +0200
+++ src/CMakeLists.txt	2015-09-17 17:54:39.730730516 +0200
@@ -379,7 +379,7 @@
 
 install (TARGETS yoshimi RUNTIME DESTINATION bin)
 
-install (DIRECTORY ../banks DESTINATION share/yoshimi
+install (DIRECTORY ../banks DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -390,7 +390,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../examples DESTINATION share/yoshimi
+install (DIRECTORY ../examples DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -401,7 +401,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../presets DESTINATION share/yoshimi
+install (DIRECTORY ../presets DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/yoshimi
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -412,7 +412,7 @@
         WORLD_READ WORLD_EXECUTE
 )
 
-install (DIRECTORY ../doc/ DESTINATION share/doc/yoshimi
+install (DIRECTORY ../doc/ DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/doc/yoshimi-${YOSHIMI_VERSION}
     FILE_PERMISSIONS
         OWNER_READ OWNER_WRITE
         GROUP_READ GROUP_WRITE
@@ -424,15 +424,15 @@
 )
 
 install (FILES ${CMAKE_CURRENT_BINARY_DIR}/yoshimi.desktop
-    DESTINATION share/applications)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/applications)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.png
-    DESTINATION share/pixmaps)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pixmaps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.svg
-    DESTINATION share/icons)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi_alt.svg
-    DESTINATION share/icons)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/icons/hicolor/scalable/apps)
 install (FILES ${CMAKE_CURRENT_SOURCE_DIR}/../desktop/yoshimi.appdata.xml
-    DESTINATION share/appdata)
+    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/appdata)
 
 set_directory_properties (PROPERTIES
     ADDITIONAL_MAKE_CLEAN_FILES "${CMAKE_SOURCE_DIR}/*UI.c* ${CMAKE_SOURCE_DIR}/src/*UI.h"
