From 1752969589b5376841b5226c3efa67f58ad83750 Mon Sep 17 00:00:00 2001
From: Moritz Bunkus <moritz@bunkus.org>
Date: Mon, 27 Mar 2017 19:27:00 +0200
Subject: [PATCH] configure: use AC_CHECK_TOOL for `strings`

Fixes #1923.
---
 ac/endianess.m4      | 4 ++--
 ac/initialization.m4 | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/ac/endianess.m4 b/ac/endianess.m4
index f232f8f..0c0d201 100644
--- a/ac/endianess.m4
+++ b/ac/endianess.m4
@@ -35,11 +35,11 @@ EOF
           if ${CC-cc} -c conftest.c -o conftest.o >>config.log 2>&1 \
               && test -f conftest.o
           then
-            if test "`strings conftest.o | grep BIGenDianSyS`"
+            if test "`"$STRINGS" conftest.o | grep BIGenDianSyS`"
             then
               ac_cv_c_bigendian="yes"
             fi
-            if test "`strings conftest.o | grep LiTTleEnDian`"
+            if test "`"$STRINGS" conftest.o | grep LiTTleEnDian`"
             then
               ac_cv_c_bigendian="no"
             fi
diff --git a/ac/initialization.m4 b/ac/initialization.m4
index 9d3ba28..5e407a5 100644
--- a/ac/initialization.m4
+++ b/ac/initialization.m4
@@ -14,6 +14,7 @@ AC_CHECK_TOOL(RANLIB, ranlib, :)
 AC_CHECK_TOOL(STRIP, strip, :)
 AC_CHECK_TOOL(AR, ar, :)
 AC_CHECK_TOOL(LD, ld, :)
+AC_CHECK_TOOL(STRINGS, strings, :)
 PKG_PROG_PKG_CONFIG
 
 dnl Check for headers
