From 8261d09ca162e00abb4783f98ccf7c726526c8d1 Mon Sep 17 00:00:00 2001
From: Moritz Bunkus <moritz@bunkus.org>
Date: Sat, 7 Jan 2017 16:29:15 +0100
Subject: [PATCH] Rakefile: fix building with rake v12.0.0 (part of Ruby 2.4.0)

Fixes #1851.
---
 Rakefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Rakefile b/Rakefile
index 10c8d68..8cd23e4 100644
--- a/Rakefile
+++ b/Rakefile
@@ -231,7 +231,9 @@ namespace :apps do
 end
 
 # Store compiler block for re-use
-cxx_compiler = lambda do |t|
+cxx_compiler = lambda do |*args|
+  t = args.first
+
   create_dependency_dirs
 
   source = t.source ? t.source : t.prerequisites.first
